kv/txn_coord_sender.go

```diff
@@ -347,61 +304,84 @@ func (tc *TxnCoordSender) maybeBeginTxn(header *proto.RequestHeader) {
 //
 // On success, and if the call is part of a transaction, the affected
 // key range is recorded as live intents for eventual cleanup upon
-// transaction commit. Upon successful txn commit, initiates cleanup
-// of intents.
-func (tc *TxnCoordSender) sendOne(ctx context.Context, call proto.Call) {
+// transaction commit, in which case they are included in the
+// EndTransaction request sent by the client.
+func (tc *TxnCoordSender) SendBatch(ctx context.Context, ba *proto.BatchRequest) (*proto.BatchResponse, error) {
+	tc.maybeBeginTxn(ba)
+	ba.CmdID = ba.GetOrCreateCmdID(tc.clock.PhysicalNow())
 	var startNS int64
-	header := call.Args.Header()
-	trace := tracer.FromCtx(ctx)
+
+	// This is the earliest point at which the request has a ClientCmdID and/or
+	// TxnID (if applicable). Begin a Trace which follows this request.
+	trace := tc.tracer.NewTrace(ba)
+	defer trace.Finalize()
+	// TODO(tschottdorf): always "Batch"
+	defer trace.Epoch(fmt.Sprintf("sending %s", ba.Method()))()
+	ctx = tracer.ToCtx(ctx, trace)
+
+	// TODO(tschottdorf): No looping through the batch will be necessary once
```

Comment:
what's the status of these TODOs? should i be reviewing with the assumption that this is ready to merge or will they be done prior to merging?